src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseExportService.java

```diff
@@ -339,24 +337,24 @@ private void deleteReposDownloadProjectRootDirectory(ProgrammingExercise program
         }
     }
 
-    private List<Repository> downloadRepositories(ProgrammingExercise programmingExercise, String targetPath) {
+    private List<Repository> downloadRepositories(ProgrammingExercise programmingExercise, String targetPath, int minimumScore, int minimumSize) {
         List<Repository> downloadedRepositories = new ArrayList<>();
-        programmingExercise.getStudentParticipations().parallelStream().forEach(participation -> {
-            var programmingExerciseParticipation = (ProgrammingExerciseParticipation) participation;
-            try {
-                if (programmingExerciseParticipation.getVcsRepositoryUrl() == null) {
-                    log.warn("Ignore participation " + participation.getId() + " for export, because its repository URL is null");
-                    return;
-                }
-                Repository repo = gitService.getOrCheckoutRepositoryForJPlag(programmingExerciseParticipation, targetPath);
-                gitService.resetToOriginMaster(repo); // start with clean state
-                downloadedRepositories.add(repo);
-            }
-            catch (GitException | GitAPIException | InterruptedException ex) {
-                log.error("clone student repository " + programmingExerciseParticipation.getVcsRepositoryUrl() + " in exercise '" + programmingExercise.getTitle()
-                        + "' did not work as expected: " + ex.getMessage());
-            }
-        });
+
+        programmingExercise.getStudentParticipations().parallelStream().filter(participation -> participation instanceof ProgrammingExerciseParticipation)
+                .map(participation -> (ProgrammingExerciseParticipation) participation).filter(participation -> participation.getVcsRepositoryUrl() != null)
+                // TODO: Filter participations by minimumSize and minimumScore
+                // .filter(...)
```

Comment:
@krusche I want to filter the `ProgrammingExerciseParticipation`s here using the options `minimumScore` and `minimumSize` sent from the plagiarism page. What is the `size` of a programming participation/submission and how can I access the latest result from the participation object?

Reply:
The minimum size is not easy to calculate. I would leave it out for programming exercises for now. We could count the lines the student has added, but this would involve some more complex git operations that we could implement in a follow up.

You can access the latest result using participation.findLatestSubmission().get().getLatestResult(). You need to make sure that in this case, those objects are properly loaded from the database. 

Another option would be to write a query to get the latest result for a participation from the database. There might be existing queries for it